packages/atclient/src/atdirectory.c

```diff
@@ -0,0 +1,152 @@
+#include "atdirectory.h"
+#include "atlogger/atlogger.h"
+#include <string.h>
+#define TAG "atdirectory"
+
+struct atdirectory_connection atdirectory_connection_create(const char *host, const uint16_t port) {
```

Comment:
Why are we going with `create` now instead of the `init` pattern?

Reply:
> Why are we going with `create` now instead of the `init` pattern?

init is commonly used to initialize a structure which has already been allocated.

create returns a struct which may allocate memory and populate values, or it can fail, in which case those values in the struct are null.

Reply:
We aren’t.

It’s private, so the developer will not have access to it.

The struct is statically allocated, only the host string is dynamically allocated at initialization